### PR TITLE
always encoding rendered result in liquid template

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 using System;
 using System.Linq;
 using System.Text;
+using System.Text.Encodings.Web;
```
Comment (stephentoub) [inline dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs]: What is this needed for?

Comment (LittleLittleCloud) [reply dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs]: unused namespace, removed.

File: dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs [hunk 2]
```diff
@@ -113,6 +113,128 @@ public async Task ItRendersUserMessagesWhenAllowUnsafeIsTrueAsync()
...
+        {
+            TemplateFormat = LiquidPromptTemplateFactory.LiquidTemplateFormat,
+            AllowUnsafeContent = true,
+            InputVariables = [
+                new() { Name = "colon", AllowUnsafeContent = true },
+                new() { Name = "encodedColon", AllowUnsafeContent = true },
+                new() { Name = "htmlTag", AllowUnsafeContent = true },
+                new() { Name = "encodedHtmlTag", AllowUnsafeContent = true },
+                new() { Name = "leftAngleBracket", AllowUnsafeContent = true },
+                new() { Name = "encodedLeftAngleBracket", AllowUnsafeContent = true }
```
Comment (markwallace-microsoft) [inline dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs]: In the Prompty case the unsafe content is the `:`. So setting `AllowUnsafeContent = true` for the variables that don't contain this isn't needed. Looking at the verified output the content looks the same.

File: dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs [hunk 3]
```diff
@@ -113,6 +113,128 @@ public async Task ItRendersUserMessagesWhenAllowUnsafeIsTrueAsync()
...
+        {
+            AllowUnsafeContent = false,
+            TemplateFormat = LiquidPromptTemplateFactory.LiquidTemplateFormat,
+            InputVariables = [
+                new() { Name = "colon", AllowUnsafeContent = false },
+                new() { Name = "encodedColon", AllowUnsafeContent = false },
+                new() { Name = "htmlTag", AllowUnsafeContent = false },
+                new() { Name = "encodedHtmlTag", AllowUnsafeContent = false },
+                new() { Name = "leftAngleBracket", AllowUnsafeContent = false },
+                new() { Name = "encodedLeftAngleBracket", AllowUnsafeContent = false }
```
Comment (markwallace-microsoft) [inline dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs]: Repeating here for clarity:
In the Prompty case the unsafe content is the :. So setting AllowUnsafeContent = true for the variables that don't contain this isn't needed. Looking at the verified output the content looks the same.

File: dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs [hunk 4]
```diff
@@ -143,15 +265,35 @@ public async Task ItDoesNotRendersUserMessagesWhenAllowUnsafeIsFalseAsync()
...
+
+        sb.AppendLine("------ ChatPromptParser Result ------");
+        foreach (var chat in chatHistory!)
+        {
+            // Append role
+            var role = chat.Role.ToString();
+            sb.Append(role + ":");
+            sb.Append(chat.Content);
+            sb.AppendLine();
+        }
```
Comment (markwallace-microsoft) [inline dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs]: The display in the verified text file is confusing.

How about JSON serialising the chat history and adding that. I think that would provide a clearer indication of what the expected output is.

File: dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs [hunk 5]
```diff
@@ -246,13 +388,16 @@ public void ReturnSomething()
...
-            <message role='user'>{{safe_input}}</message>
+            system:
+            This is the system message
+            user:
+            {{unsafe_input}}
+            user:
+            {{safe_input}}
             """;
 
         var kernel = new Kernel();
```
Comment (markwallace-microsoft) [inline dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs:404]: Can you also test with this template

```csharp
        var template =
            """
            system:
            This is the system message
            user:
            ```csharp
            /// <summary>
            /// Example code with comment in the system prompt
            /// </summary>
            public void ReturnSomething()
            {
            	// no return
            }
            ```
            """;
```

Here is the expected behaviour https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.UnitTests/PromptTemplate/KernelPromptTemplateTests.cs#L851-L888

Comment (LittleLittleCloud) [reply dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs:404]: This test case has already added, check here

https://github.com/LittleLittleCloud/semantic-kernel/blob/73905073206810717f6bfed6a0aaf53cb042de1a/dotnet/src/Extensions/PromptTemplates.Liquid.UnitTests/LiquidTemplateTest.cs#L344

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**